### DISCUSSION TITLE
求助

### DISCUSSION THREAD

**Discussion by hyk1st** [criyle/go-judge#51] [Q&A] [answered]
你好，我想问一下我开了grpc端口，但是沙箱显示连不上要怎么解决呀

采用命令：
docker run -it --rm --privileged --shm-size=256m -p 5050:5050 criyle/executorserver -enable-grpc

报错：
root@VM-12-6-ubuntu:/simpken/demo/go-judge# ./executorshell 
2022/11/27 01:12:39 ExecStream rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:5051: connect: connection refused"

**Comment by hyk1st** [comment]
docker run之后报了这个错
{"level":"info","ts":1669487472.960998,"caller":"executorserver/main.go:281","msg":"create 1 prefork containers"}
2022/11/26 18:31:12 prefork environment failed  container: failed to start container fork/exec /proc/self/exe: operation not permitted

**Comment by hyk1st** [comment]
你好，我跑通了，请问文件这部分要怎么测试，就是怎么创建输入输出，我按照样例跑，需要先做什么呀，

显示：
[
	{
		"status": "Internal Error",
		"exitStatus": 0,
		"error": "execve: start: execve: no such file or directory",
		"time": 311840,
		"memory": 131072,
		"runTime": 0,
		"files": {
			"stderr": "",
			"stdout": ""
		},
		"fileIds": {
			"a.cc": "GPKOGGDQ"
		},
		"fileError": [
			{
				"name": "a",
				"type": "CopyOutOpen",
				"message": "open a: no such file or directory"
			}
		]
	}
]

**Comment by criyle** [comment, marked as answer]
如果是使用容器启动，需要在容器内安装 g++

**Reply by hyk1st** [reply]
大佬我可以加一下你的联系方式吗，我最近在写一个小型的oj练练手，然后遇到一些问题想请教你

**Reply by hyk1st** [reply]
就是大佬我还不理解那个评测时的输入文件，和标答文件要怎么传入和使用

**Reply by criyle** [reply]
可以看看 https://github.com/criyle/go-judger-demo

**Reply by hyk1st** [reply]
你好，我想问下沙箱终端一直运行不了怎么办啊，报错：
2022/11/27 01:12:39 ExecStream rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:5051: connect: connection refused"

**Reply by criyle** [reply]
需要 5051 端口 docker run -it --rm --privileged --shm-size=256m -p 5050:5050 -p 5051:5051 criyle/executorserver -enable-grpc

**Reply by hyk1st** [reply]
谢谢大佬

**Reply by hyk1st** [reply]
大佬，我跑通了，最后问一下，就是每次编译完产生的文件需要自己调用/file去删除对吗，然后还有就是，这个评测系统是不是没有标答比对，要自己在服务端对stdout和标答进行比对呀

**Reply by criyle** [reply]
是的

**Reply by hyk1st** [reply]
谢谢大佬
